test/keyboardEvents.js

```diff
@@ -0,0 +1,114 @@
+			var listener = sinon.stub();
+			var element = document.createElement('input');
+			dom.enterDocument(element);
+			dom.on(element, 'keyup-enter', listener);
+			dom.triggerEvent(element, 'keyup', {keyCode: KEYMAP.ENTER});
+			assert.strictEqual(1, listener.callCount);
+		});
+
+		it('should not trigger the listener to an unmatched key alias', function() {
+			var listener = sinon.stub();
+			var element = document.createElement('input');
+			dom.enterDocument(element);
+			dom.on(element, 'keyup-enter', listener);
+			dom.triggerEvent(element, 'keyup', {keyCode: KEYMAP.SPACE});
+			assert.strictEqual(0, listener.callCount);
+		});
+
+		it('should not stop listening to an unparameterized keyboard event', function() {
+			var listener = sinon.stub();
+			var element = document.createElement('input');
+			dom.enterDocument(element);
+			dom.on(element, 'keyup', listener);
+			dom.triggerEvent(element, 'keyup', {keyCode: KEYMAP.ENTER});
+			assert.strictEqual(1, listener.callCount);
+		});
+
+		it('should pass through the event object the custom event name used', function(done) {
+			var element = document.createElement('input');
+			dom.enterDocument(element);
+			dom.on(element, 'keyup-enter', function(event) {
+				assert.strictEqual(event.customType, 'keyup-enter');
+				done();
+			});
+			dom.triggerEvent(element, 'keyup', {keyCode: KEYMAP.ENTER});
+		});
+	});
+
+	describe('From IncrementalDOM', function() {
+		it('should listen to the keyboard event only if the key pressed matches with the event name using (onKeydown-enter) pattern', function() {
+			var callback = sinon.stub();
+			class TestComponent extends Component {
+				render() {
+					IncDom.elementOpen('div');
+					IncDom.elementVoid('input', null, null, 'onKeydown-enter', 'handleClick');
+					IncDom.elementClose('div');
+				}
+			}
+			TestComponent.RENDERER = IncrementalDomRenderer;
+			TestComponent.prototype.handleClick = callback;
+
+			var component = new TestComponent();
+			dom.triggerEvent(component.element.querySelector('input'), 'keydown', {keyCode: KEYMAP.ENTER});
+			assert.strictEqual(KEYMAP.ENTER, callback.args[0][0].keyCode, 'The keyCode should match with the event name');
+		});
+
+		it('should listen to the keyboard event only if the key pressed matches with the event name using (data-on-eventname) pattern', function() {
+			var callback = sinon.stub();
+			class TestComponent extends Component {
+				render() {
+					IncDom.elementOpen('div');
+					IncDom.elementVoid('input', null, null, 'data-onkeydown-enter', 'handleClick');
+					IncDom.elementClose('div');
+				}
+			}
+			TestComponent.RENDERER = IncrementalDomRenderer;
+			TestComponent.prototype.handleClick = callback;
+
+			var component = new TestComponent();
+			dom.triggerEvent(component.element.querySelector('input'), 'keydown', {keyCode: KEYMAP.ENTER});
+			assert.strictEqual(KEYMAP.SPACE, callback.args[0][0].keyCode, 'The keyCode should match with the event name');
```

Comment:
This test is failing on my side, and it makes sense that it'd fail. This should be checking that the key code is equal to `KEYMAP.ENTER`, not `KEYMAP.SPACE`, right?